source/trx2junit/Models/Test.cs

```diff
@@ -5,9 +5,9 @@ namespace trx2junit.Models
 {
     public class Test
     {
-        public Times                            Times           { get; set; }
-        public ResultSummary                    ResultSummary   { get; set; }
-        public Dictionary<Guid, TestDefinition> TestDefinitions { get; set; } = new Dictionary<Guid, TestDefinition>();
-        public Dictionary<Guid, UnitTestResult> UnitTestResults { get; set; } = new Dictionary<Guid, UnitTestResult>();
+        public Times                                         Times { get; set; }
+        public ResultSummary                                 ResultSummary { get; set; }
+        public Dictionary<Guid, TestDefinition>              TestDefinitions { get; set; } = new Dictionary<Guid, TestDefinition>();
+        public Dictionary<Guid, ICollection<UnitTestResult>> UnitTestResults { get; set; } = new Dictionary<Guid, ICollection<UnitTestResult>>();
```

Comment:
Let me sleep on this.

Not sure if I should change this class

Reply:
Reason I changed this, is so we can look up the `UnitTestResults` for a given `TestId` Key.

However it feels wrong doing it in this class.

I think I may be better to create an `ILookup` or something in the `JUnitParser`

Reply:
> I think I may be better to create an `ILookup` or something in the `JUnitBuilder`

Agreed. This logic should be outside of `Test`-class.

Reply:
Maybe we could model this `1:n` relationship in `TestDefinition` like
```c#
using System;
using System.Collections.Generic;

namespace trx2junit
{
    public class TestDefinition
    {
        public Guid   Id          { get; set; }
        public Guid   ExecutionId { get; set; }
        public string TestClass   { get; set; }
        public string TestMethod  { get; set; }

        public ICollection<UnitTestResult> TestResults { get; set; } = new HashSet<UnitTestResult>();
    }
}
```

And let the `TrxParser` do the work.

What do you think?

Reply:
Not sure we need this.

Up to you if you want it modelled this way.

I have a solution which I'm happier with now.

Reply:
Your solution looks pretty good and simple 👍 We'll keep it this way.